sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -172,6 +174,22 @@ public DefaultAzureCredentialBuilder managedIdentityResourceId(String resourceId
         return this;
     }
 
+    /**
+     * Specifies the object ID of user assigned or system assigned identity, when this credential is running
+     * in an environment with managed identities. If unset, the value in the AZURE_CLIENT_ID environment variable
+     * will be used. If neither is set, the default value is null and will only work with system assigned
+     * managed identities and not user assigned managed identities.
+     *
+     * Only one of managedIdentityResourceId and managedIdentityClientId can be specified.
+     *
+     * @param objectId the object ID
+     * @return the DefaultAzureCredentialBuilder itself
+     */
+    public DefaultAzureCredentialBuilder managedIdentityObjectId(String objectId) {
+        this.managedIdentityObjectId = objectId;
+        return this;
+    }
```

Comment:
I think we decided to only put this on `ManagedIdentityCredential` - check with @ahsonkhan

Reply:
scoped to MI cred only.